com.unity.shadergraph/Editor/GraphDeltaRegistry/Registry/RegistryPlaceholder.cs

```diff
@@ -146,40 +146,27 @@ public ShaderFunction GetShaderFunction(INodeReader data, ShaderContainer contai
         class PowNode : Defs.INodeDefinitionBuilder
         {
             public RegistryKey GetRegistryKey() => new RegistryKey { Name = "Pow", Version = 1 };
+
             public RegistryFlags GetRegistryFlags() => RegistryFlags.Func;
 
-            /**
-             * BuildNode defines the input and output ports of a node, along with
-             * the port types.
-             */
-            public void BuildNode(INodeReader userData, INodeWriter nodeWriter, Registry registry)
+            public void BuildNode(NodeHandler node, Registry registry)
             {
-                var portWriter = nodeWriter.AddPort<GraphType>(userData, "In", true, registry);
-                portWriter.SetField<int>(GraphType.kLength, 1);
-                portWriter = nodeWriter.AddPort<GraphType>(userData, "Exp", true, registry);
-                portWriter.SetField<int>(GraphType.kLength, 1);
-                portWriter = nodeWriter.AddPort<GraphType>(userData, "Out", false, registry);
-                portWriter.SetField<int>(GraphType.kLength, 1);
+                var port = node.AddPort("In", true, true);
+                port.AddField(GraphType.kLength, 1);
+                port = node.AddPort("Exp", true, true);
+                port.AddField(GraphType.kLength, 1);
+                port = node.AddPort("Out", false, true);
+                port.AddField(GraphType.kLength, 1);
```

Comment:
Does this do the same GraphType setup that AddPort<T> did before?

Reply:
This is my mistake. I can change it.